crawl-ref/source/viewmap.cc

```diff
@@ -975,12 +1004,38 @@ level_pos map_follow_stairs(bool up, const coord_def &pos)
 
 void process_map_command(command_type cmd)
 {
-    // XX cleaner API for this
-    shared_ptr<ui::Widget> l = ui::top_layout();
-    if (UIMapView *mv = dynamic_cast<UIMapView *>(l.get()))
-        mv->process_command(cmd);
+    if (map_view)
+        map_view->process_command(cmd);
+}
+
+#ifdef USE_TILE_LOCAL
+void start_far_viewing(coord_def viewed_location)
+{
+    if (map_view)
+        map_view->start_far_viewing(viewed_location);
+}
+
+void stop_far_viewing()
+{
+    if (map_view)
+        map_view->stop_far_viewing();
 }
 
+bool is_far_viewing()
+{
+    if (!map_view)
```

Comment:
Can condense the function into a single line

`return map_view && map_view->is_far_viewing()`

Reply:
Done. That's definitely nicer to read